### DISCUSSION TITLE
Relationship(joinedload) unexpected behaviour

### DISCUSSION THREAD

**Discussion by levsh** [sqlalchemy/sqlalchemy#10263] [Usage Questions] [answered]
Hi!
I have a parent/child relationship and try to load all children for the parent. But parent loaded for child also and this is unexpected for me.
My test code:
```python
import pytest
import sqlalchemy as sa
from sqlalchemy.ext.asyncio import async_sessionmaker
from sqlalchemy.orm import DeclarativeBase, relationship

class Base(DeclarativeBase):
    pass

class Parent(Base):
    __tablename__ = "parents"
    __mapper_args__ = {"eager_defaults": False}

    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
    children = relationship("Child", uselist=True, viewonly=True)

class Child(Base):
    __tablename__ = "children"
    __mapper_args__ = {"eager_defaults": False}

    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
    parent_id = sa.Column(sa.Integer, sa.ForeignKey("parents.id"), nullable=False)
    parent = relationship("Parent", uselist=False, viewonly=True)

@pytest.mark.asyncio
async def test_relationship(engine):
    async with engine.begin() as conn:
        await conn.run_sync(Base.metadata.create_all)

    async_session = async_sessionmaker(engine, expire_on_commit=False)

    async with async_session() as sess:
        async with sess.begin():
            parent = Parent()
            sess.add(parent)
            await sess.flush()
            child = Child(parent_id=parent.id)
            sess.add(child)

    async with async_session() as sess:
        async with sess.begin():
            stmt = sa.select(Parent).where(Parent.id == 1).options(sa.orm.joinedload(Parent.children))
            parent = (await sess.execute(stmt)).unique().scalars().one()

        assert parent.children  # this is ok

        assert parent.children[0].parent  # this is not ok, expect parent.children{i}.parent not loaded

        print()
        print(parent.children[0].parent)
```
Output is:
```
 <tests.test_relationship.Parent object at 0x7f9c6168a190>
```
I expect parent.childred[0].parent is not loaded. This is right behaviour or bug?
```
asyncpg                    0.28.0
SQLAlchemy                 2.0.20
Python                     3.11.4
```

**Comment by zzzeek** [comment, marked as answer]
hi -

`parent.children[0].parent ` is `parent`, SQLAlchemy saw that Child.parent is a many-to-one and simply retrieved it from the [identity map](https://docs.sqlalchemy.org/en/20/glossary.html#term-identity-map).  No additional SQL was emitted; turn on [echoing](https://docs.sqlalchemy.org/en/20/core/engines.html#dbengine-logging) and you will see the SQL is only the lead row + the joinedload:

```
SELECT parents.id, children_1.id AS id_1, children_1.parent_id 
FROM parents LEFT OUTER JOIN children AS children_1 ON parents.id = children_1.parent_id 
WHERE parents.id = ?

```

**Reply by levsh** [reply]
@zzzeek thank you for the answer.
Is there a way to tell Alchemy not to retrieve it from identity map?
Reason, i try to load Alchemy model to Pydantic model and get recursion error.

**Reply by zzzeek** [reply]
you can make it raise an error, but that would also fail since the error is not an `AttributeError`.

you can use the "noload" strategy to make it return `None`, but now you've made a `relatoinship()` that returns garbage information - why have it at all then ?

if pydantic still in 2023 does not support the ability to omit attributes, why not simply remove the relationship altogether?

**Reply by zzzeek** [reply]
also have you considered posting an issue with pydantic?  this seems like a pretty serious bug on their end.   Serializers should always be aware that loops can be present in object graphs, and guarding against this is trivial.
